dubbo-common/src/main/java/org/apache/dubbo/common/URL.java

```diff
@@ -1058,6 +1039,19 @@ public URL addParameters(Map<String, String> parameters) {
         return new URL(protocol, username, password, host, port, path, map);
     }
 
+    /**
+     * test if parameters are already contained in the url
+     */
+    private boolean contains(Map<String, String> parameters) {
+        for (Map.Entry<String, String> entry : parameters.entrySet()) {
+            String value = getParameters().get(entry.getKey());
+            if (!StringUtils.isEquals(value, entry.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }

```

Comment:
can we not use just 

`return parameters.equals(getParameters()));`

what do you say?

Reply:
i think meaning of the original code is contain: 
the parameters field may contains key that doesn't exists in the given parameters

Reply:
i have put it in the pr #3458